test/models/edition_test.rb

```diff
@@ -198,6 +198,21 @@ def draft_second_edition_from(published_edition)
   end
 
 # test cloning into different edition types
+  test "it should be possible to clone from any type to any type" do
+    # Note that the new edition won't be valid in cases where the new type has
+    # required fields (such as licence_identifier on Licences). This is OK
+    # because when Publisher saves the clone, it already skips validations. The
+    # user will then be required to populate those values before they save the
+    # edition again.
+    Edition.subclasses.permutation(2).each do |source_class, destination_class|
```

Comment:
I kinda feel like having this loop define the tests would be good, then it will generate individual tests like `"it should be possible to clone from SmartAnswer to Licence"` etc.